Source/WebCore/css/CSSAttrValue.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2024 Alexsander Borges Damaceno <alexbdamac@gmail.com>. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "config.h"
+#include "CSSAttrValue.h"
+
+#include "CSSPrimitiveValue.h"
+
+namespace WebCore {
+
+Ref<CSSAttrValue> CSSAttrValue::create(String attributeName, RefPtr<CSSValue>&& fallback)
+{
+    return adoptRef(*new CSSAttrValue(WTFMove(attributeName), WTFMove(fallback)));
+}
+
+bool CSSAttrValue::equals(const CSSAttrValue& other) const
+{
+    RefPtr fallback = m_fallback;
+    RefPtr otherFallback = other.m_fallback;
+    return m_attributeName == other.m_attributeName && fallback == otherFallback;
```

Comment:
You can just compare directly the fallbacks without storing them, given the `==` operator should just be consulting trivial getters

Reply:
Totally drive-by, but pretty sure that comparing the fallback RefPtrs doesn't do what you want, as it only does a shallow (pointer) comparison: https://searchfox.org/wubkat/rev/1e937703de6af995ef2cd93867292c9da1d3b86c/Source/WTF/wtf/RefPtr.h#198-202

Reply:
Emilio is right, can we compare the fallback `stringValue()` ?